clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -536,7 +544,9 @@ static KafkaAdminClient createInternal(AdminClientConfig config, TimeoutProcesso
                 time,
                 true,
                 apiVersions,
-                logContext);
+                null,
+                logContext,
+                (hostResolver == null) ? new DefaultHostResolver() : hostResolver);
```

Comment:
Why the ternary here when you could've passed `new DefaultHostResolver()` on line 489?

Reply:
Not a big problem I think. I just followed the pattern that additional args in overloaded methods are nullable and, if null a suitable value is used within the implememntation. this is what already happens with the `TimeoutProcessorFactory` for example